# List Comprehensions Lab

Complete the following set of exercises to solidify your knowledge of list comprehensions.

In [78]:
import os
import numpy as np
import pandas as pd


### 1. Use a list comprehension to create and print a list of consecutive integers starting with 1 and ending with 50.

In [16]:
lista = [e for e in range(1,51)]
print(lista)

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]


### 2. Use a list comprehension to create and print a list of even numbers starting with 2 and ending with 200.

In [17]:
lista = [e for e in range(1,201) if e%2 == 0]
print(lista)

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200]


### 3. Use a list comprehension to create and print a list containing all elements of the 10 x 4 Numpy array below.

In [19]:
a = np.array([[0.84062117, 0.48006452, 0.7876326 , 0.77109654],
       [0.44409793, 0.09014516, 0.81835917, 0.87645456],
       [0.7066597 , 0.09610873, 0.41247947, 0.57433389],
       [0.29960807, 0.42315023, 0.34452557, 0.4751035 ],
       [0.17003563, 0.46843998, 0.92796258, 0.69814654],
       [0.41290051, 0.19561071, 0.16284783, 0.97016248],
       [0.71725408, 0.87702738, 0.31244595, 0.76615487],
       [0.20754036, 0.57871812, 0.07214068, 0.40356048],
       [0.12149553, 0.53222417, 0.9976855 , 0.12536346],
       [0.80930099, 0.50962849, 0.94555126, 0.33364763]])

lista = [f for e in a for f in e]
print(lista)


[0.84062117, 0.48006452, 0.7876326, 0.77109654, 0.44409793, 0.09014516, 0.81835917, 0.87645456, 0.7066597, 0.09610873, 0.41247947, 0.57433389, 0.29960807, 0.42315023, 0.34452557, 0.4751035, 0.17003563, 0.46843998, 0.92796258, 0.69814654, 0.41290051, 0.19561071, 0.16284783, 0.97016248, 0.71725408, 0.87702738, 0.31244595, 0.76615487, 0.20754036, 0.57871812, 0.07214068, 0.40356048, 0.12149553, 0.53222417, 0.9976855, 0.12536346, 0.80930099, 0.50962849, 0.94555126, 0.33364763]


### 4. Add a condition to the list comprehension above so that only values greater than or equal to 0.5 are printed.

In [20]:
lista2 = [f for e in a for f in e if f>0.5]
print(lista2)

[0.84062117, 0.7876326, 0.77109654, 0.81835917, 0.87645456, 0.7066597, 0.57433389, 0.92796258, 0.69814654, 0.97016248, 0.71725408, 0.87702738, 0.76615487, 0.57871812, 0.53222417, 0.9976855, 0.80930099, 0.50962849, 0.94555126]


### 5. Use a list comprehension to create and print a list containing all elements of the 5 x 2 x 3 Numpy array below.

In [46]:
b = np.array([[[0.55867166, 0.06210792, 0.08147297],
        [0.82579068, 0.91512478, 0.06833034]],

       [[0.05440634, 0.65857693, 0.30296619],
        [0.06769833, 0.96031863, 0.51293743]],

       [[0.09143215, 0.71893382, 0.45850679],
        [0.58256464, 0.59005654, 0.56266457]],

       [[0.71600294, 0.87392666, 0.11434044],
        [0.8694668 , 0.65669313, 0.10708681]],

       [[0.07529684, 0.46470767, 0.47984544],
        [0.65368638, 0.14901286, 0.23760688]]])

In [41]:
#for element in b:
#    for column in element:
#        for value in column:
#            print(value)

lista3 = [value for element in b for column in element for value in column]
print(lista3)


[0.55867166, 0.06210792, 0.08147297, 0.82579068, 0.91512478, 0.06833034, 0.05440634, 0.65857693, 0.30296619, 0.06769833, 0.96031863, 0.51293743, 0.09143215, 0.71893382, 0.45850679, 0.58256464, 0.59005654, 0.56266457, 0.71600294, 0.87392666, 0.11434044, 0.8694668, 0.65669313, 0.10708681, 0.07529684, 0.46470767, 0.47984544, 0.65368638, 0.14901286, 0.23760688]


### 5. Add a condition to the list comprehension above so that the last value in each subarray is printed, but only if it is less than or equal to 0.5.

In [77]:
lista_3 = []
#for e in b:
#    if e[0][2] <= 0.5:
#        print(e[0])
#        list_3.append(e[0])
#    if e[1][2] <= 0.5:
#        print(e[1])
#        list_3.append(e[1])
#print('RESULT--------\n',list_3)
#for a in b:
#    for c in a:
#        print(c[2])
#            if c[2] <= 0,5:
lista_3 = [c[2] for a in b for c in a if c[2]<= 0.5]
print(lista_3)

[0.08147297, 0.06833034, 0.30296619, 0.45850679, 0.11434044, 0.10708681, 0.47984544, 0.23760688]


### 6. Use a list comprehension to select and print the names of all CSV files in the */data* directory.

In [127]:
os.chdir('/home/alex/Documentos/Iron/Clases/datamad1019/module-1/lab-list-comprehensions/data')
list_dir = os.listdir('/home/alex/Documentos/Iron/Clases/datamad1019/module-1/lab-list-comprehensions/data')
'''
for element in list_dir:
        if element[-1] == 'v':
            print(element)

'''
list_6 = [element for element in list_dir if element[-1] == 'v']
print(list_6)

['sample_file_2.csv', 'sample_file_1.csv', 'sample_file_5.csv', 'sample_file_0.csv', 'sample_file_8.csv', 'sample_file_3.csv', 'sample_file_7.csv', 'sample_file_6.csv', 'sample_file_9.csv', 'sample_file_4.csv']


### 7. Use a list comprehension and the Pandas `read_csv` and `concat` methods to read all CSV files in the */data* directory and combine them into a single data frame. Display the top 10 rows of the resulting data frame.

In [132]:
os.chdir('/home/alex/Documentos/Iron/Clases/datamad1019/module-1/lab-list-comprehensions/data')


read_doc = [pd.read_csv(document) for document in list_dir]
concated_doc = [pd.concat(read_doc,ignore_index=True)]
print(concated_doc)


#for document in list_dir:
#    read_doc.append(pd.read_csv(document)
#concated_doc = [pd.concat(read_doc,ignore_index=True)]
#print(concated_doc)
    

[           0         1         2         3         4         5         6  \
0   0.948664  0.215285  0.918270  0.599951  0.755120  0.971609  0.103190   
1   0.163236  0.803926  0.916655  0.775234  0.644890  0.701362  0.910208   
2   0.934136  0.031410  0.954057  0.853387  0.642160  0.681184  0.317198   
3   0.757038  0.918964  0.475459  0.837686  0.149645  0.819032  0.611996   
4   0.263455  0.816283  0.336707  0.587997  0.285871  0.619942  0.018027   
5   0.276827  0.260054  0.942397  0.113187  0.781355  0.475740  0.152061   
6   0.995885  0.158381  0.244274  0.962163  0.651900  0.930665  0.577190   
7   0.641917  0.821055  0.392437  0.782617  0.510762  0.428320  0.017324   
8   0.806532  0.569258  0.148175  0.809987  0.459632  0.735762  0.730664   
9   0.311185  0.501165  0.365979  0.782807  0.776795  0.797199  0.791946   
10  0.437114  0.156941  0.183148  0.817785  0.747632  0.726855  0.944758   
11  0.213337  0.503490  0.802663  0.443484  0.700893  0.129329  0.007176   
12  0.62569

### 8. Use a list comprehension to select and print the column numbers for columns from the data set whose median is less than 0.48.

### 9. Use a list comprehension to add a new column (20) to the data frame whose values are the values in column 19 minus 0.1. Display the top 10 rows of the resulting data frame.

### 10. Use a list comprehension to extract and print all values from the data set that are between 0.7 and 0.75.